housekeeper/store/api/handlers/read.py

```diff
@@ -249,3 +243,33 @@ def get_all_non_archived_files(self, tag_names: list[str]) -> list[File]:
             tag_names=tag_names,
             is_archived=False,
         ).all()
+
+    def get_archives(
+        self, archival_task_id: int = None, retrieval_task_id: int = None
+    ) -> Optional[list[File]]:
+        """Returns all entries in the archive table with the specified archival/retrieval task id."""
+        if not archival_task_id and not retrieval_task_id:
+            return self._get_query(table=Archive).all()
+        if archival_task_id and retrieval_task_id:
+            return apply_archive_filter(
+                archives=apply_archive_filter(
+                    archives=self._get_query(table=Archive),
+                    filter_functions=[ArchiveFilter.FILTER_BY_ARCHIVING_TASK_ID],
+                    task_id=archival_task_id,
+                ),
+                filter_functions=[ArchiveFilter.FILTER_BY_RETRIEVAL_TASK_ID],
+                task_id=retrieval_task_id,
+            ).all()
+        if archival_task_id:
+            return apply_archive_filter(
+                archives=self._get_query(table=Archive),
+                filter_functions=[ArchiveFilter.FILTER_BY_ARCHIVING_TASK_ID],
+                task_id=archival_task_id,
+            ).all()
+
+        else:
```

Comment:
```suggestion
```
I do not think it is needed sice you return in all ifs

Reply:
Yes - I just mean that if you will execute this block with or without the else if the execution got this far, but maybe I am missing something.

Reply:
Oh! Now I see what you mean! Good catch!